### PR TITLE
fix(providers): change type of EmailConfig

### PR DESCRIPTION
**Pull request by nkg168**
## Reasoning 💡

This change allow to set the [well-known service](https://github.com/nodemailer/nodemailer/blob/master/lib/well-known/services.json) in the mail settings.

Types for argument of createTransport is here.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/nodemailer/index.d.ts#L54

## Checklist 🧢

- ~[ ] Documentation~
- ~[ ] Tests~
- [x] Ready to be merged

### CODE REVIEW

File: packages/next-auth/src/providers/email.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import { createTransport } from "nodemailer"
 
 import type { CommonProviderOptions } from "."
-import type { Options as SMTPConnectionOptions } from "nodemailer/lib/smtp-connection"
+import type { Options as SMTPTransportOptions } from "nodemailer/lib/smtp-transport"
```
Comment (lluia) [inline packages/next-auth/src/providers/email.ts:4]: Does this type exist? I can't see it on **DefinitivelyTyped**:
https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/nodemailer/lib

Are we use both types have precisely the same options? We don't want this to be a breaking change 💅🏽

**Review by 0ubbe — CHANGES_REQUESTED** [review]

File: packages/next-auth/src/providers/email.ts (hunk 1 above)
Comment (nkg168) [reply packages/next-auth/src/providers/email.ts:4]: this type is here.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/nodemailer/lib/smtp-transport.d.ts#L42

Options named SMTPTransportOptions extends Options named SMTPConnectionOptions.

Comment (lluia) [reply packages/next-auth/src/providers/email.ts:4]: I see, so this type [`SMTPTransportOptions`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/nodemailer/lib/smtp-transport.d.ts#L42-L46) extends [`SMTPConnectionOptions`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/nodemailer/lib/smtp-connection/index.d.ts#L131-L172) with few more things 🤔 

In that case it's ok for me 👍🏽

**Review by 0ubbe — APPROVED** [review]

**Comment by ThangHuuVu** [comment]
@lluia shall we merge this? 😄

**Pull request merged**